src/services/keystroke.js

```diff
@@ -5,8 +5,18 @@
 
 Controller.open(function(_) {
   _.keystroke = function(key, evt) {
+    if(this.keystrokeHandlers) {
+      for(var i = 0; i < this.keystrokeHandlers.length; i++)
+        this.keystrokeHandlers[i].call(this, key, evt);
+    }
     this.cursor.parent.keystroke(key, evt, this);
   };
+  _.registerKeystrokeHandler = function(fn) {
+    if(!this.keystrokeHandlers) {
+      this.keystrokeHandlers = []
```

Comment:
**opinion** Do we know when the start of the lifecycle is? What is `this` and when is it created? I would be nicer to initialise `this. keystrokeHandlers` at construction and not have to test if it exists all the time.

Reply:
The lifecycle here is a bit tricky and it would need some modification to the core controller code. We'd have to create a new method like `_.initKeystrokeHandlers` and the call that from when the controller is initialized